src/elements/content-sidebar/activity-feed/task-new/AssigneeList.js

```diff
@@ -0,0 +1,162 @@
+    [TASK_NEW_REJECTED]: messages.tasksFeedStatusRejected,
+    [TASK_NEW_COMPLETED]: messages.tasksFeedStatusCompleted,
+    [TASK_NEW_NOT_STARTED]: null,
+};
+
+const Datestamp = ({ date }: { date: ISODate | Date }) => {
+    return <ReadableTime timestamp={new Date(date).getTime()} alwaysShowTime relativeThreshold={0} />;
+};
+
+const AvatarDetails = React.memo(({ user, status, completedAt, className }) => {
+    const statusMessage = statusMessages[status] || null;
+    return (
+        <div className={className}>
+            <div className="bcs-AssigneeList-detailsName">{user.name}</div>
+            {statusMessage && completedAt && (
+                <div className="bcs-AssigneeList-detailsStatus">
+                    <FormattedMessage {...statusMessage} values={{ dateTime: <Datestamp date={completedAt} /> }} />
+                </div>
+            )}
+        </div>
+    );
+});
+
+class AssigneeList extends React.Component<Props, State> {
+    state = {
+        isCollapsed: true,
+    };
+
+    static defaultProps = {
+        minAssignees: MIN_ASSIGNEES_SHOWN,
+        users: {},
+    };
+
+    listTitleId = uniqueId('assignee-list-title-');
+
+    showLessAssignees = () => {
+        this.setState({
+            isCollapsed: true,
+        });
+    };
+
+    showMoreAssignees = async () => {
+        const { users } = this.props;
+
+        if (users.next_marker) {
+            try {
+                await this.props.onExpand();
+            } catch (err) {
+                // do nothing
+            }
+        }
+
+        this.setState({
+            isCollapsed: false,
+        });
+    };
+
+    render() {
+        const { minAssignees, users, getAvatarUrl } = this.props;
+        const { isCollapsed } = this.state;
+        const { entries = [], next_marker } = users;
+        const entryCount = entries.length;
+        const hiddenAssigneeCount = Math.max(0, entryCount - minAssignees);
+        const numVisibleAssignees = isCollapsed ? minAssignees : entryCount;
+        const visibleUsers = entries
+            .slice(0, numVisibleAssignees)
+            .map(({ id, target, status, completed_at: completedAt }) => {
+                return (
+                    <li key={id} className="bcs-AssigneeList-listItem">
+                        <AvatarGroupAvatar
+                            status={status}
+                            className="bcs-AssigneeList-listItemAvatar"
+                            user={target}
+                            getAvatarUrl={getAvatarUrl}
+                        />
+                        <AvatarDetails
+                            className="bcs-AssigneeList-listItemDetails"
+                            user={target}
+                            status={status}
+                            completedAt={completedAt}
+                        />
+                    </li>
+                );
+            });
+
+        const maxAdditionalAssignees = TASKS_PAGE_SIZE - minAssignees;
```

Comment:
Can the API not return more than 20?

Reply:
The initial page returns 20 max (should also return `limit` so we can calculate dynamically once a bug is fixed)